runcoverage.sh

```diff
@@ -23,40 +23,67 @@ then
   fi
 fi
 
-$NUGET install -Verbosity quiet -OutputDirectory packages -Version 4.6.519 OpenCover
+$NUGET install -Verbosity quiet -OutputDirectory packages -Version 2017.1.20170613.162720 JetBrains.dotCover.CommandLineTools
 $NUGET install -Verbosity quiet -OutputDirectory packages -Version 2.4.5.0 ReportGenerator
-OPENCOVER=$PWD/packages/OpenCover.4.6.519/tools/OpenCover.Console.exe
+
+DOTCOVER=$PWD/packages/JetBrains.dotCover.CommandLineTools.2017.1.20170613.162720/tools/dotCover.exe
 REPORTGENERATOR=$PWD/packages/ReportGenerator.2.4.5.0/tools/ReportGenerator.exe
 FIND=/usr/bin/find
 
-rm -rf coverage
-mkdir coverage
+if [ ! -d coverage ]
+then
+ mkdir coverage
+fi
 
 while read csproj
 do
   testdir=$(dirname $csproj)
   api=$(basename $(realpath $testdir/..))
+
+  [ ! -d "$testdir/bin/Release/net452" ] && [ ! -d "$testdir/bin/Release/netcoreapp1.0" ] && continue
+
   echo "Running coverage for $api"
-  # OpenCover is picky about how it finds the excluded files. There may be a better approach than this,
-  # but it'll do for now.
-  generatedFiles=$($FIND $testdir/../$api -name '*.cs' | grep -v "obj/" | xargs -n 1 grep -l "// Generated" | sed 's/.*\/.*\//*\\\\*\\\\/g' | tr '\n' ';')
-  echo Coverage excluding files $generatedFiles
-  
-  $OPENCOVER \
-    -target:"c:\Program Files\dotnet\dotnet.exe" \
-    -targetargs:"test --no-build -f net452 -c Release $csproj" \
-    -mergeoutput \
-    -hideskipped:File \
-    -output:coverage/coverage.xml \
-    -filter:"+[$api]*" \
-    -oldStyle \
-    -searchdirs:$testdir/bin/$CONFIG/net452 \
-    -register:user \
-    -excludebyfile:$generatedFiles
 
+  echo "<CoverageParams>" > "coverage/$api.xml"
+  echo " <TargetExecutable>/Program Files/dotnet/dotnet.exe</TargetExecutable>" >> "coverage/$api.xml"
+if [ -d "$testdir/bin/Release/net452" ] ; then
+  echo " <TargetArguments>test --no-build -f net452 -c Release</TargetArguments>" >> "coverage/$api.xml"
+else
+  echo " <TargetArguments>test --no-build -f dotnet -c Release</TargetArguments>" >> "coverage/$api.xml"
+fi
+  echo " <TargetWorkingDir>../$testdir</TargetWorkingDir>" >> "coverage/$api.xml"
+  echo " <Output>$api.dvcr</Output>" >> "coverage/$api.xml"
+  echo " <LogFile>$api.log</LogFile>" >> "coverage/$api.xml"
+  echo "</CoverageParams>" >> "coverage/$api.xml"
+
+  $DOTCOVER cover coverage/$api.xml
 done < AllTests.txt
 
+echo "<MergeParams>" > "coverage/merge.xml"
+for filename in coverage/*.dvcr; do
+  echo "<Source>$(basename $filename)</Source>" >> "coverage/merge.xml"
+done
+echo "<Output>all.dvcr</Output>" >> "coverage/merge.xml"
+echo "</MergeParams>" >> "coverage/merge.xml"
+
+$DOTCOVER merge coverage/merge.xml
+
+echo "<ReportParams>" > "coverage/report.xml"
+echo "<Source>all.dvcr</Source>" >> "coverage/report.xml"
+echo "<Output>coverage.xml</Output>" >> "coverage/report.xml"
+echo "<ReportType>DetailedXML</ReportType>" >> "coverage/report.xml"
+echo "</ReportParams>" >> "coverage/report.xml"
+
+$DOTCOVER report coverage/report.xml
```

Comment:
Just use command line parameters for this and maybe merge? Building an XML file for the cover command seems reasonable (and is useful for per-API ad hoc testing), but I think we can probably do merge/report more readably without the XML.